tools/clang/test/CodeGenSPIRV/type.constant-buffer.assign.hlsl

```diff
@@ -0,0 +1,32 @@
+// RUN: %dxc -T vs_6_0 -E main
+
+// CHECK:      OpName %type_ConstantBuffer_S "type.ConstantBuffer.S"
+// CHECK-NEXT: OpMemberName %type_ConstantBuffer_S 0 "someFloat"
+// CHECK-NEXT: OpMemberName %type_ConstantBuffer_S 1 "another"
+
+// CHECK:      OpDecorate %type_ConstantBuffer_S Block
+
+// CHECK: %type_ConstantBuffer_S = OpTypeStruct %float %float
+
+struct S
+{
+    float someFloat;
+    float another;
+};
+
+// CHECK: %_ptr_Uniform_type_ConstantBuffer_S = OpTypePointer Uniform %type_ConstantBuffer_S
+
+// CHECK: %S = OpTypeStruct %float %float
+// CHECK: %_ptr_Function_S = OpTypePointer Function %S
+
+// CHECK: %buffer = OpVariable %_ptr_Uniform_type_ConstantBuffer_S Uniform
+ConstantBuffer<S> buffer;
+
+void main()
+{
+// CHECK:           %local = OpVariable %_ptr_Function_S Function
+
+// CHECK: [[val:%\d+]] = OpLoad %type_ConstantBuffer_S %buffer
+// CHECK:                OpStore %local [[val]]
```

Comment:
We are very selective about the types of invalid code that we are willing to generate. I'm wondering if in this case we are missing a call to 

https://github.com/microsoft/DirectXShaderCompiler/blob/15c87234584676574ca4a4ebebde03aab9822994/tools/clang/lib/SPIRV/SpirvEmitter.h#L203-L207

Can you see if you can modify the generated code to have it store a value of the correct type?

Reply:
It turns out that this kind of invalid code was already being generated for `ConstantBuffer` and `TextureBuffer` in some cases; I believe these cases no longer require validation.